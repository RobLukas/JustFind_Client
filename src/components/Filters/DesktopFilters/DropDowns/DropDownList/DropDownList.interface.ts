type ButtonComponentProps = {
  handleClick(): void;
};

interface DropDownListProps {
  ButtonComponent: React.ComponentType<ButtonComponentProps>;
  collection: string[];
}

export default DropDownListProps;
