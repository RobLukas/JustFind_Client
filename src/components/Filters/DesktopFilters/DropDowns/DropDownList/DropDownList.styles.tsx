import styled from 'styled-components';

const DropDownList = styled.div`
  position: relative;
`;

const ContentWrapper = styled.div`
  position: absolute;
  background: ${({ theme }) => theme.styles.secondary.background};
  box-shadow: 0px 2px 24px 0px rgba(0, 0, 0, 0.5);
  border-radius: 4px;
  margin-top: 5px;
  left: 50%;
  transform: translate(-50%, 0);
  z-index: 2;
`;

const ListWrapper = styled.ul`
  margin: 0;
  min-width: 100px;
  padding: 8px 0 8px 0;
  white-space: nowrap;
  line-height: 1.5;
`;

const ListItem = styled.li`
  text-align: center;
  justify-content: center;
  font-size: 14px;
  display: flex;
  color: ${({ theme }) => theme.styles.filter.button.color};
  cursor: pointer;
  margin: 0;
  padding: 7px 24px 7px 24px;
  &:hover {
    background: ${({ theme }) => theme.styles.dropdown.button.hover};
  }
`;

export default {
  DropDownList,
  ContentWrapper,
  ListWrapper,
  ListItem,
};
