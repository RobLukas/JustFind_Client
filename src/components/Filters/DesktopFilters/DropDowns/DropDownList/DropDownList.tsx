import React, { FC, useCallback, useState } from 'react';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import { v4 as uuid } from 'uuid';

import S from './DropDownList.styles';
import DropDownListProps from './DropDownList.interface';

const DropDownList: FC<DropDownListProps> = ({
  ButtonComponent,
  collection,
}: DropDownListProps) => {
  const [open, setOpen] = useState(false);
  const handleClick = useCallback(() => setOpen(!open), [open]);
  const handleClickAway = useCallback(() => setOpen(false), []);

  return (
    <S.DropDownList>
      <ButtonComponent handleClick={handleClick} />
      {open ? (
        <ClickAwayListener onClickAway={handleClickAway}>
          <S.ContentWrapper>
            <S.ListWrapper>
              {collection.map((item) => (
                <S.ListItem key={uuid()}>{item}</S.ListItem>
              ))}
            </S.ListWrapper>
          </S.ContentWrapper>
        </ClickAwayListener>
      ) : null}
    </S.DropDownList>
  );
};

export default DropDownList;
