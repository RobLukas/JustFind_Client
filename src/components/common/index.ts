export { default as JobTitle } from './JobTitle/JobTitle';
export { default as LogoCompany } from './LogoCompany/LogoCompany';
export { default as NoJobOfferNotify } from './NoJobOffersNotify/NoJobOfferNotify';
export { default as Salary } from './Salary/Salary';

// Buttons
export { default as DotsButton } from './Buttons/DotsButton/DotsButton';
export { default as SelectButton } from './Buttons/SelectButton/SelectButton';
export { default as CategoryButton } from './Buttons/CategoryButton/CategoryButton';
export { default as SwitchModeButton } from './Buttons/SwitchModeButton/SwitchModeButton';
export { default as TechnologyIconButton } from './Buttons/TechnologyIconButton/TechnologyIconButton';
export { default as SalaryButton } from './Buttons/SalaryButton/SalaryButton';
export { default as ExperienceButton } from './Buttons/ExperienceButton/ExperienceButton';
