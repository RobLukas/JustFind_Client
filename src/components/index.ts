// Header
export { default as Header } from './Header/Header';

// Offers
export { default as OfferDetails } from './OfferDetails/OfferDetails';

// Map
export { default as MarkerMap } from './Map/Marker/Marker';
