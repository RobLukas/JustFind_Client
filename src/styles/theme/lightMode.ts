export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
