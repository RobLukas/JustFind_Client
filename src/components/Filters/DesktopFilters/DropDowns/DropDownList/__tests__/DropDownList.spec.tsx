import React from 'react';
import 'jest-styled-components';
import { cleanup, render } from '@testing-library/react';

import { DotsButton } from 'components/common';
import lightMode from 'styles/theme/lightMode';
import DropDownList from '../DropDownList';
import { ThemeProvider } from 'styled-components';

describe('<DropDownList />', () => {
  afterEach(cleanup);
  it('render correcly', () => {
    const collection = ['1', '2', '3'];
    const { container } = render(
      <ThemeProvider theme={lightMode}>
        <DropDownList ButtonComponent={DotsButton} collection={collection} />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
