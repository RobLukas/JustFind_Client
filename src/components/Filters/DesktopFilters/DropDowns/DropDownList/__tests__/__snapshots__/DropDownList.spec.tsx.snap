// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropDownList /> render correcly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-basis: 35px;
  -ms-flex-preferred-size: 35px;
  flex-basis: 35px;
  height: 35px;
}

.c2 {
  display: block;
}

.c0 {
  position: relative;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root c2"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 10c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm12 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm-6 0c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
